File: backend-go/internal/modules/system/controller/print_template_controller.go
package controller

import (
	"drip-admin/backend-go/internal/common"
	"drip-admin/backend-go/internal/modules/system/dto"
	"github.com/gin-gonic/gin"
)

var (
	_ = common.ApiResponse{}
	_ = dto.PrintTemplateSaveRequest{}
)

// PrintTemplates returns print templates.
// @Summary List print templates
// @Tags print-template
// @Produce json
// @Security ApiKeyAuth
// @Param page query int false "Page"
// @Param pageSize query int false "Page size"
// @Param code query string false "Code"
// @Param name query string false "Name"
// @Param status query int false "Status"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/print-template [get]
func (ctl *Controller) PrintTemplates(c *gin.Context) {
	ctl.svc.PrintTemplates(c)
}

// PrintTemplate returns a print template detail.
// @Summary Print template detail
// @Tags print-template
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Print template ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/print-template/{id} [get]
func (ctl *Controller) PrintTemplate(c *gin.Context) {
	ctl.svc.PrintTemplate(c)
}

// CreatePrintTemplate creates a print template.
// @Summary Create print template
// @Tags print-template
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param request body dto.PrintTemplateSaveRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/print-template [post]
func (ctl *Controller) CreatePrintTemplate(c *gin.Context) {
	ctl.svc.CreatePrintTemplate(c)
}

// CopyPrintTemplate copies a print template.
// @Summary Copy print template
// @Tags print-template
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Print template ID"
// @Param request body dto.PrintTemplateCopyRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/print-template/{id}/copy [post]
func (ctl *Controller) CopyPrintTemplate(c *gin.Context) {
	ctl.svc.CopyPrintTemplate(c)
}

// UpdatePrintTemplate updates a print template.
// @Summary Update print template
// @Tags print-template
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Print template ID"
// @Param request body dto.PrintTemplateSaveRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/print-template/{id} [put]
func (ctl *Controller) UpdatePrintTemplate(c *gin.Context) {
	ctl.svc.UpdatePrintTemplate(c)
}

// DeletePrintTemplate deletes a print template.
// @Summary Delete print template
// @Tags print-template
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Print template ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/print-template/{id} [delete]
func (ctl *Controller) DeletePrintTemplate(c *gin.Context) {
	ctl.svc.DeletePrintTemplate(c)
}

// PrintTemplateStatus updates a print template status.
// @Summary Update print template status
// @Tags print-template
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Print template ID"
// @Param request body dto.StatusUpdateRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/print-template/{id}/status [put]
func (ctl *Controller) PrintTemplateStatus(c *gin.Context) {
	ctl.svc.PrintTemplateStatus(c)
}

File: backend-python/app/modules/system/controller/print_template_controller.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.common.api_response import ApiResponse, success
from app.common.pagination import PageQuery
from app.modules.system.dto.print_template_request import PrintTemplateCopyRequest, PrintTemplateSaveRequest
from app.modules.system.dto.user_request import StatusUpdateRequest
from app.modules.system.router import get_db, require_permission
from app.modules.system.service.print_template_service import PrintTemplateService

router = APIRouter(tags=["print-template"])


def service(db: AsyncSession = Depends(get_db)) -> PrintTemplateService:
    return PrintTemplateService(db)


@router.get("/print-template", response_model=ApiResponse)
async def print_templates(
    page: PageQuery = Depends(),
    code: str | None = None,
    name: str | None = None,
    status: int | None = None,
    _: dict = Depends(require_permission("system:printTemplate:list")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    return success(await current.list_templates(page, code, name, status))


@router.get("/print-template/{id}", response_model=ApiResponse)
async def print_template_detail(
    id: int,
    _: dict = Depends(require_permission("system:printTemplate:list")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    return success(await current.detail(id))


@router.post("/print-template", response_model=ApiResponse)
async def create_print_template(
    body: PrintTemplateSaveRequest,
    _: dict = Depends(require_permission("system:printTemplate:create")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    return success(await current.save(None, body))


@router.post("/print-template/{id}/copy", response_model=ApiResponse)
async def copy_print_template(
    id: int,
    body: PrintTemplateCopyRequest,
    _: dict = Depends(require_permission("system:printTemplate:create")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    return success(await current.copy(id, body))


@router.put("/print-template/{id}", response_model=ApiResponse)
async def update_print_template(
    id: int,
    body: PrintTemplateSaveRequest,
    _: dict = Depends(require_permission("system:printTemplate:update")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    await current.save(id, body)
    return success()


@router.delete("/print-template/{id}", response_model=ApiResponse)
async def delete_print_template(
    id: int,
    _: dict = Depends(require_permission("system:printTemplate:delete")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    await current.delete(id)
    return success()


@router.put("/print-template/{id}/status", response_model=ApiResponse)
async def print_template_status(
    id: int,
    body: StatusUpdateRequest,
    _: dict = Depends(require_permission("system:printTemplate:update")),
    current: PrintTemplateService = Depends(service),
) -> ApiResponse:
    await current.update_status(id, body.status)
    return success()
